rest-api-model-ql/src/main/kotlin/org/modelix/sample/restapimodelql/LightModelClientWrapper.kt

```diff
@@ -101,4 +115,54 @@ class LightModelClientWrapper(
         return this.lightModelClient.Area()
     }
 
+
+
+    suspend fun updateRoom(newRoom: Room){
+        var result: N_Room? = null
+        val root = lightModelClient.waitForRootNode()
+        if (root != null) {
+
+            var decodedReference = URLDecoder.decode(newRoom.roomRef, Charset.defaultCharset())
+            val result = resolveNodeIdToConcept(decodedReference) as N_Room
+            // test these
+            // "the right way"
+            //INodeReferenceSerializer.deserialize(actualRef).resolveNode(lightModelClient.getRootNode()?.getArea())
+
+            // warning not performant!
+            // root.typed<N_Repository>().descendants(true).ofType<N_Room>()
+
+            lightModelClient.runWrite {
+                result.name = newRoom.name
+                result.maxPlaces = newRoom.maxPlaces
+                result.hasRemoteEquipment = newRoom.hasRemoteEquipment!!
+            }
+        }
+    }
+    suspend fun updateLecture(newLecture: Lecture){
+        println("UPDATE LECTURE")
+        var result: N_Lecture? = null
+        val root = lightModelClient.waitForRootNode()
+        if (root != null) {
+
+            var decodedReference = URLDecoder.decode(newLecture.lectureRef, Charset.defaultCharset())
+            val result = resolveNodeIdToConcept(decodedReference) as N_Lecture
+
+            var decodedRoomReference = URLDecoder.decode(newLecture.room, Charset.defaultCharset())
+            val roomRefConcept = resolveNodeIdToConcept(decodedRoomReference) as N_Room
+            // TODO: test ways to resolve concept
+            // "the right way"
+            //INodeReferenceSerializer.deserialize(actualRef).resolveNode(lightModelClient.getRootNode()?.getArea())
+
+            // warning this is not performant!
+            // root.typed<N_Repository>().descendants(true).ofType<N_Room>()
+
+            lightModelClient.runWrite {
+                result.name = newLecture.name
+                result.description = newLecture.description
+                result.room = roomRefConcept
+                result.maxParticipants = newLecture.maxParticipants
+            }
+        }
```

Comment:
```suggestion
        val root = lightModelClient.waitForRootNode() ?: return

        val decodedLectureReference = URLDecoder.decode(newLecture.lectureRef, Charset.defaultCharset())
        val result = resolveNodeIdToConcept(decodedLectureReference) as N_Lecture

        val decodedRoomReference = URLDecoder.decode(newLecture.room, Charset.defaultCharset())
        val roomRefConcept = resolveNodeIdToConcept(decodedRoomReference) as N_Room
        // TODO: test ways to resolve concept
        // "the right way"
        //INodeReferenceSerializer.deserialize(actualRef).resolveNode(lightModelClient.getRootNode()?.getArea())

        // warning this is not performant!
        // root.typed<N_Repository>().descendants(true).ofType<N_Room>()

        lightModelClient.runWrite {
            result.name = newLecture.name
            result.description = newLecture.description
            result.room = roomRefConcept
            result.maxParticipants = newLecture.maxParticipants
        }
```
if can be shortened with ?: and the references can be val. rename decodedReference to decodedLectureReference